cmd/frontend/graphqlbackend/insights.graphql

```
@@ -587,6 +599,177 @@ input LineChartSearchInsightInput {
     viewControls: InsightViewControlsInput
 }
 
+"""
+Input for the data series allowed on line charts.
+"""
+input LineChartDataSeriesInput {
+    """
+    Array of series ids in the user defined order.
+    """
+    order: [ID!]!
+
+    """
+    Array of search generated data series.
+    """
+    search: [SearchDataSeriesInput!]
+
+    # Other types go here.. could potentially have a separate type for compute?
```

@CristinaBirkel @coury-clark if I understood this properly, when we will have another insight type in that PR's system we have to update all mutations input and add a special optional field for that newly added insight type. 

This might be a problem because in a system when we're getting more and more chart visualisation and more insight types it would be a combination explosion when it comes to adding another chart or insight type. 

Let say 
- we are adding new insight type (we have to update all mutation of all existing chart types and add optional field)
- we are adding new chart type (we have to create a mutation where we absorb all existing insight types that we currently have) 

In both cases it seems like a lot of work just for adding addition insight or chart type IMO.

---

One thing to note here is that the reason for having different input types per chart type is just to enforce type safety. We could use one type if we don't care about that. (By type safety here, I mean enforcing which chart types are allowed to have which data series. So that we don't need to return a runtime error if there is a mismatch.)

In other words, a new series type may or may not be added to all chart types, depending on what kind of charts it is allowed on.